### ISSUE TITLE
ceph (nautilus): OSD crashloops due to panic on unexportable field in cluster resource

### ISSUE THREAD

**Issue by d-luu** [rook/rook#3166]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
OSDs CrashLooping
```
rook-ceph          rook-ceph-osd-0-74cbb648b9-g945l                         0/1     CrashLoopBackOff   10         26m
rook-ceph          rook-ceph-osd-1-5f4896c55b-9dmj7                         0/1     CrashLoopBackOff   10         26m
rook-ceph          rook-ceph-osd-2-559884d97f-mkwrs                         0/1     Error              10         26m
rook-ceph          rook-ceph-osd-prepare-rookfix-worker-01-xc27q            0/2     Completed          0          25m
rook-ceph          rook-ceph-osd-prepare-rookfix-worker-02-lsqtk            0/2     Completed          0          25m
rook-ceph          rook-ceph-osd-prepare-rookfix-worker-03-jwscf            0/2     Completed          0          25m
```
In the rook operator logs, you can see that a panic is thrown:
```
2019-05-11 17:34:40.860735 I | op-cluster: added finalizer to cluster rook-ceph
E0511 17:34:40.863208       8 runtime.go:67] Observed a panic: cannot handle unexported field: {v1.ClusterSpec}.Resources["mgr"].Limits["cpu"].i
consider using AllowUnexported or cmpopts.IgnoreUnexported
/home/rook/go/src/github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/home/rook/go/src/github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/home/rook/go/src/github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:522
/usr/local/go/src/runtime/panic.go:513
/home/rook/go/src/github.com/rook/rook/vendor/github.com/google/go-cmp/cmp/options.go:208
/home/rook/go/src/github.com/rook/rook/vendor/github.com/google/go-cmp/cmp/compare.go:307
...
```

**Expected behavior:**
OSDs running

**How to reproduce it (minimal and precise):**
* Set resource requests and limits for Rook components in cluster.yaml
* Set rook cross cluster related stuff namespace to rook-ceph-system and cluster components to rook-ceph (not sure if its needed)
* Deploy using Rook 1.0 with Ceph v14.2.1-20190430.

**Environment**:
* OS (e.g. from /etc/os-release): SUSE Linux Enterprise Server 12 SP3
* Kernel (e.g. `uname -a`): 4.4.162-94.72
* Cloud provider or hardware configuration: vSphere
* Rook version (use `rook version` inside of a Rook Pod): 1.0
* Kubernetes version (use `kubectl version`): 1.12.3
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):

**Comment by travisn** [comment]
Reopening for backport to 1.0

**Comment by travisn** [comment]
Already included with v1.0.2
